dbms/src/Interpreters/ExternalLoader.cpp

```diff
@@ -443,8 +515,9 @@ void ExternalLoader::reload(const std::string & name)
 
 ExternalLoader::LoadablePtr ExternalLoader::getLoadableImpl(const std::string & name, bool throw_on_error) const
 {
-    std::lock_guard lock{map_mutex};
+    ensureReloadFinished(name, throw_on_error);
```

Comment:
Discussed that here we will wait until all dictionaries will be reloaded (in case ExternalLoader::reload() was called).

Reply:
done
I've written a non-blocking version of `ensureReloadFinished()`.